src/Piece.js

```diff
@@ -1,5 +1,47 @@
 /* Represents a piece in the game */
 class Piece 
 {
+	constructor()
+	{
+		this.moveVectors = [];
+		this.captureVectors = [];
+		this.moveCaptureVectors = [];
+		this.player = undefined;
+	}
 	
-}
\ No newline at end of file
+	static Create(baseObj)
+	{
+		/* Currently assumes baseObj has a string for
+		 * move, capture, moveCapture, and player.
+		 */
+		return new Piece()
+			.setMoveVectors(Vector.Create(baseObj.move))
+			.setCaptureVectors(Vector.Create(baseObj.capture))
+			.setMoveCaptureVectors(Vector.Create(baseObj.moveCapture))
+			.setPlayer(baseObj.player);
+	}
+	
+	setMoveVectors(newValue)
```

Comment:
The name `moveVectors` seems more suitable for the argument. It's hard to justify now, but if these methods grew to be more complicated, a more specific name would make things easier.